rpcs3/util/asm.hpp

```diff
@@ -435,6 +435,15 @@ namespace utils
 		return factor1 > 0 && T{umax} / factor1 < factor2 ? T{umax} : static_cast<T>(factor1 * factor2);
 	}
 
+	inline void trigger_write_page_fault(void* ptr)
+	{
+#if defined(ARCH_X64) && !defined(_MSC_VER)
+		__asm__ volatile("lock orl $0, 0(%0)" :: "r" (ptr));
+#else
+		*static_cast<volatile u32*>(ptr) |= 0;
```

Comment:
This needs to be atmomic, it doesnt matter that it is 0

Reply:
I've changed it back to atomic_t<u32> but this operation can and will be optimized away by clang on other architectures(ie arm). Ideally someone should put an ifdef for arm here.